File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Trackers/DoneDone/Domain/DoneDoneIssueBase.cs
using System;

namespace bugTrapKit.DoneDone
{

	public class DoneDoneIssueBase : DoneDoneSimpleItem
	{
		public int OrderNumber { get; set; }

		// public SimpleItem Project { get; set; }

		public DoneDoneProject Project { get; set; }

		public DateTime? DueDate { get; set; }

		public DateTime CreatedOn { get; set; }

		public DoneDoneSimpleItem Status { get; set; }

		public DoneDoneSimpleItem Tester { get; set; }

		public DoneDoneSimpleItem Fixer { get; set; }

		public DoneDoneSimpleItem Priority { get; set; }

		#region Helper

		public bool HasProject {
			get {
				return Project != null && Project.Id > 0;
			}
		}

		public bool HasStatus {
			get {
				return Status != null && Status.Id > 0;
			}
		}

		public bool HasTester {
			get {
				return Tester != null && Tester.Id > 0;
			}
		}

		public bool HasFixer {
			get {
				return Fixer != null && Fixer.Id > 0;
			}
		}

		public bool HasPriority {
			get {
				return Priority != null && Priority.Id > 0;
			}
		}

		public bool HasDueDate {
			get {
				return DueDate.HasValue;
			}
		}

		#endregion
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Trackers/DoneDone/Domain/DoneDoneIssueBase.swift
//
//  DoneDoneIssueBase.swift
//  bugTrap
//
//  Created by Colby L Williams on 8/25/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation

class DoneDoneIssueBase : DoneDoneSimpleItem {
	
	var orderNumber: Int?
	
	// var project: Project?
	
	var project = DoneDoneProject()
	
	// var project: SimpleItem?
	
	var dueDate: NSDate?
	
	var createdOn: NSDate?
	
	var status = DoneDoneSimpleItem()
	
	var tester = DoneDoneSimpleItem()
	
	var fixer = DoneDoneSimpleItem()
	
	var priority = DoneDoneSimpleItem()	
}